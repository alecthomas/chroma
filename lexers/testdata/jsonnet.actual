local application = 'my-app';
local module = 'uwsgi_module';
local dir = '/var/www';
local permission = 644;

{
  'uwsgi.ini': std.manifestIni({
    sections: {
      uwsgi: {
        module: module,
        pythonpath: dir,
        socket: dir + '/uwsgi.sock',
        'chmod-socket': permission,
        callable: application,
        logto: '/var/log/uwsgi/uwsgi.log',
      },
    },
  }),

  'init.sh': |||
    #!/usr/bin/env bash
    mkdir -p %(dir)s
    touch %(dir)s/initialized
    chmod %(perm)d %(dir)s/initialized
  ||| % {dir: dir, perm: permission},

  'cassandra.conf': std.manifestYamlDoc({
    cluster_name: application,
    seed_provider: [
      {
        class_name: 'SimpleSeedProvider',
        parameters: [{ seeds: '127.0.0.1' }],
      },
    ],
  }),
}
