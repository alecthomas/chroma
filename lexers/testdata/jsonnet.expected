[
  {"type":"Keyword","value":"local"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"application"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'my-app'"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"local"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"module"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'uwsgi_module'"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"local"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"dir"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'/var/www'"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"local"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"permission"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberFloat","value":"644"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameVariable","value":"'uwsgi.ini'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"std.manifestIni"},
  {"type":"Punctuation","value":"({"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameVariable","value":"sections"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n      "},
  {"type":"NameVariable","value":"uwsgi"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"module"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"module"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"pythonpath"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"dir"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"socket"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"dir"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'/uwsgi.sock'"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"'chmod-socket'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"permission"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"callable"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"application"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"logto"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'/var/log/uwsgi/uwsgi.log'"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n      "},
  {"type":"Punctuation","value":"},"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Punctuation","value":"},"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Punctuation","value":"}),"},
  {"type":"TextWhitespace","value":"\n\n  "},
  {"type":"NameVariable","value":"'init.sh'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"|||\n    #!/usr/bin/env bash\n    mkdir -p %(dir)s\n    touch %(dir)s/initialized\n    chmod %(perm)d %(dir)s/initialized\n  |||"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"%"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"NameVariable","value":"dir"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"dir"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"perm"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"permission"},
  {"type":"Punctuation","value":"},"},
  {"type":"TextWhitespace","value":"\n\n  "},
  {"type":"NameVariable","value":"'cassandra.conf'"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"std.manifestYamlDoc"},
  {"type":"Punctuation","value":"({"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameVariable","value":"cluster_name"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"application"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameVariable","value":"seed_provider"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"TextWhitespace","value":"\n      "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"class_name"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'SimpleSeedProvider'"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":"\n        "},
  {"type":"NameVariable","value":"parameters"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"[{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"seeds"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"'127.0.0.1'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"}],"},
  {"type":"TextWhitespace","value":"\n      "},
  {"type":"Punctuation","value":"},"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Punctuation","value":"],"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Punctuation","value":"}),"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"}
]
